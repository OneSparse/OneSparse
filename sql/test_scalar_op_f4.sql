-- # Scalar `float4`
--
-- Test various scalar math operations with native Postgres types
select (1::float4)::scalar + -1::float4;
select 1::float4 + (-1::float4)::scalar;
select (1::float4)::scalar - 1::float4;
select 1::float4 - (1::float4)::scalar;
select (1::float4)::scalar * -1::float4;
select 1::float4 * (-1::float4)::scalar;
select (1::float4)::scalar / 1::float4;
select 1::float4 / (1::float4)::scalar;

-- Test construction of min, zero and max values:
select '-1'::float4::scalar;
select '0'::float4::scalar;
select '1'::float4::scalar;

-- Test setting a scalar value from max to 2
select set('1'::float4::scalar, 2);

-- Test various casting functions used by the CREATE CAST machinery:
select scalar_float4((-1)::float4);
select scalar_float4((0)::float4);
select scalar_float4((1)::float4);
select float4_scalar((-1)::float4::scalar);
select float4_scalar((0)::float4::scalar);
select float4_scalar((1)::float4::scalar);

-- These casting functions cast the Postgres type `float4` to the
-- GraphBLAS scalar type `GrB_INT32`.

select cast(-1::float4 as scalar);
select cast(0::float4 as scalar);
select cast(1::float4 as scalar);

-- These tests cast back from the scalar type `GrB_INT32` to the
-- Postgres type `float4`

select cast((-1::float4)::scalar as float4);
select cast((0::float4)::scalar as  float4);
select cast((1::float4)::scalar as scalar);
