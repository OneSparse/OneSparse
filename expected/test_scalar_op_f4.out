-- # Scalar 'float4' Operators
--
-- This documentation is also tests for the code, the examples below
-- show the literal output of these statements from Postgres.
--
-- Some setup to make sure warnings are shown, and that the extension
-- is installed.
set client_min_messages = 'WARNING';
create extension if not exists onesparse;
-- Test various scalar math operations with native Postgres types
select (1::float4)::scalar + -1::float4;
  ?column?   
-------------
 f4:0.000000
(1 row)

select 1::float4 + (-1::float4)::scalar;
 ?column? 
----------
        0
(1 row)

select (1::float4)::scalar - 1::float4;
  ?column?   
-------------
 f4:0.000000
(1 row)

select 1::float4 - (1::float4)::scalar;
 ?column? 
----------
        0
(1 row)

select (1::float4)::scalar * -1::float4;
   ?column?   
--------------
 f4:-1.000000
(1 row)

select 1::float4 * (-1::float4)::scalar;
 ?column? 
----------
       -1
(1 row)

select (1::float4)::scalar / 1::float4;
  ?column?   
-------------
 f4:1.000000
(1 row)

select 1::float4 / (1::float4)::scalar;
 ?column? 
----------
        1
(1 row)

